app/javascript/components/Course/GithubRepos/CourseGithubRepoProjectReposStatistics.jsx

```diff
@@ -0,0 +1,524 @@
+                if (! (team in team_stats)) {
+                    team_stats[team] = {issueEdits: 0, timelineItems: 0, total: 0}
+                }
+                team_stats[team]["issueEdits"] += this.state.edit_combined_count[team]
+                team_stats[team]["total"] += this.state.edit_combined_count[team]
+            }
+        );
+        Object.keys(this.state.timeline_combined_count).forEach( 
+            (team) => {
+                if (! (team in team_stats)) {
+                    team_stats[team] = {issueEdits: 0, timelineItems: 0, total: 0}
+                }
+                team_stats[team]["timelineItems"] += this.state.timeline_combined_count[team]
+                team_stats[team]["total"] += this.state.timeline_combined_count[team]
+            }
+        );
+
+        const team_stats_vector = objectToVector(team_stats,"name");
+
+        this.setState({ 
+            team_stats: team_stats,
+            team_stats_vector: team_stats_vector
+        });
+    }
+
+   
+    computeOverallStudentStats = () => {
+        const databaseIds = Object.keys(this.props.databaseId_to_student)
+        let student_stats = {}
+        Object.keys(this.state.edit_combined_count_by_student).forEach( 
+            (databaseId) => {
+                if (! (databaseId in student_stats)) {
+                    student_stats[databaseId] = {issueEdits: 0, timelineItems: 0, total: 0}
+                }
+                student_stats[databaseId]["issueEdits"] += this.state.edit_combined_count_by_student[databaseId]
+                student_stats[databaseId]["total"] += this.state.edit_combined_count_by_student[databaseId]
+            }
+        );
+        Object.keys(this.state.timeline_combined_count_by_student).forEach( 
+            (databaseId) => {
+                if (! (databaseId in student_stats)) {
+                    student_stats[databaseId] = {issueEdits: 0, timelineItems: 0, total: 0}
+                }
+                student_stats[databaseId]["timelineItems"] += this.state.timeline_combined_count_by_student[databaseId]
+                student_stats[databaseId]["total"] += this.state.timeline_combined_count_by_student[databaseId]
+            }
+        );
+
+        Object.keys(student_stats).forEach(
+            (databaseId) => {
+                student_stats[databaseId]["team"]= (databaseId in this.props.databaseId_to_team) ?
+                    this.props.databaseId_to_team[databaseId] :
+                    "N/A";
+                if (databaseId in this.props.databaseId_to_student) {
+                    let student = this.props.databaseId_to_student[databaseId];
+                    student_stats[databaseId]["login"] = student.login;
+                    student_stats[databaseId]["name"] = student.name;
+
+                } else {
+                    student_stats[databaseId]["login"] = "N/A";
+                    student_stats[databaseId]["name"] = "N/A";
+                }
+            }
+        );
```

Comment:
I wonder if some of these _clean data from the API_ functions could be factored out? These seem to have the highest chance of being reused.